### DISCUSSION TITLE
How to list or check all valid option for a field.

### DISCUSSION THREAD

**Discussion by nomad-cyanide** [netbox-community/netbox#17896] [General]
Hi

I am trying to create an easy way for my tech-users to document a new VPN-tunnel and would like to be able to validate if a given input is valid for the field we use.

Example: When creating an IKE proposal, you have 8 options to choose from when setting "Encryption Algorithm". Is there a way to get that list through the API (preferably using pynetbox), so I can check if the users input is valid. 
I suppose I could build my own list manually,  but it doesn't seem like the clever way to do it. 

I suppose this would be useful for a lot of other fields and purposes in Netbox as well.

Is this possible?

**Comment by candlerb** [comment]
Yes, using an OPTIONS request. Documentation: https://netboxlabs.com/docs/netbox/en/stable/reference/filtering/#filtering-by-choice-field

Example:

```
# curl -sSH "Authorization: Token $TOKEN" -H "Content-Type: application/json" -H  "Accept: application/json; indent=4" \
  -X OPTIONS "$NETBOX/api/vpn/ike-proposals/" | jq .actions.POST.encryption_algorithm.choices
[
  {
    "value": "aes-128-cbc",
    "display_name": "128-bit AES (CBC)"
  },
  {
    "value": "aes-128-gcm",
    "display_name": "128-bit AES (GCM)"
  },
  {
    "value": "aes-192-cbc",
    "display_name": "192-bit AES (CBC)"
  },
  {
    "value": "aes-192-gcm",
    "display_name": "192-bit AES (GCM)"
  },
  {
    "value": "aes-256-cbc",
    "display_name": "256-bit AES (CBC)"
  },
  {
    "value": "aes-256-gcm",
    "display_name": "256-bit AES (GCM)"
  },
  {
    "value": "3des-cbc",
    "display_name": "3DES"
  },
  {
    "value": "des-cbc",
    "display_name": "DES"
  }
]
```

For pynetbox, check the docs: https://pynetbox.readthedocs.io/en/latest/endpoint.html#pynetbox.core.endpoint.Endpoint.choices

```
>>> nb.vpn.ike_proposals.choices()
{'authentication_method': [{'value': 'preshared-keys', 'display_name': 'Pre-shared keys'}, {'value': 'certificates', 'display_name': 'Certificates'},
...
>>> nb.vpn.ike_proposals.choices()['encryption_algorithm']
[{'value': 'aes-128-cbc', 'display_name': '128-bit AES (CBC)'}, {'value': 'aes-128-gcm', 'display_name': '128-bit AES (GCM)'},
...
```
